templates/react/src/App.js

```diff
@@ -83,12 +83,34 @@ export default function App() {
           {window.accountId}!
         </h1>
         <form onSubmit={async event => {
+          event.preventDefault()
+
+          // get elements from the form using their id attribute
+          const { fieldset, greeting } = event.target.elements
+
           // hold onto new user-entered value from React's SynthenticEvent for use after `await` call
-          const newGreeting = event.target.elements.greeting.value
+          const newGreeting = greeting.value
+
+          // disable the form while the value gets updated on-chain
+          fieldset.disabled = true
 
-          // fire frontend-agnostic submit behavior, including data persistence
-          // look in utils.js to see how this updates data on-chain!
-          await onSubmit(event)
+          try {
+            // make an update call to the smart contract
+            await window.contract.setGreeting({
+              // pass the value that the user entered in the greeting field
+              message: newGreeting
+            })
+          } catch (e) {
+            alert(
+              'Something went wrong! ' +
+              'Maybe you need to sign out and back in? ' +
+              'Check your browser console for more info.'
+            )
+            throw e
+          } finally {
+            // re-enable the form, whether the call succeeded or failed
+            fieldset.disabled = false
+          }
 
           // update local `greeting` variable to match persisted value
```

Comment:
I would move all this code inside the try-catch, so if there's an error, no notification other than the alert is raised

Reply:
the code in question does two things

1. update local `greeting` variable to match persisted value
2. show notification, then hide again after 10 seconds

We only want to do these things if the code in the `try` block completes _successfully_. As-written, this is exactly what it will do. 

We could put it in the `try` block below the `contract.setGreeting` call. However, this seems less straightforward to me – the `catch` & `finally` are there to deal with the `contract.setGreeting` call failing/completing. If there were several more steps in the `try` block, it would be unclear which part of it might throw the error we want to deal with.

Reply:
> If there were several more steps in the try block, it would be unclear which part of it might throw the error we want to deal with

ok, yes, you're right.